ts/features/zendesk/saga/index.ts

```diff
@@ -79,3 +99,54 @@ export function* watchZendeskSupportSaga() {
     dismissSupport();
   });
 }
+/**
+ *
+ * @param getSession is the API call to get the session tokens.
+ * The goal of this saga is to take Zendesk token from the BE in order to properly report to support.
+ */
+function* getZendeskTokenSaga(
+  getSession: ReturnType<typeof BackendClient>["getSession"]
+) {
+  try {
+    // Define the fields needed for the token request, in this case, the needed field is only 'zendeskToken'
+    const fields = formatRequestedTokenString(false, ["zendeskToken"]);
+    const isFastLogin = yield* select(isFastLoginEnabledSelector);
+
+    const response = (yield* call(
+      withRefreshApiCall,
+      getSession({ fields }),
+      getZendeskToken.failure("401") // if the error is 401 the error screen is not show thanks this parameter
+    )) as unknown as SagaCallReturnType<typeof getSession>;
+
+    if (E.isLeft(response)) {
+      throw Error(readableReport(response.left));
+    } else {
+      if (response.right.status === 200) {
+        yield* put(getZendeskToken.success());
+        const currentValues = yield* select(sessionInfoSelector);
+        yield* put(
+          sessionInformationLoadSuccess(
+            getOnlyNotAlreadyExistentValues(
+              response.right.value,
+              O.isSome(currentValues) && currentValues.value
+            )
+          )
+        );
+      } else {
+        if (!isFastLogin || response.right.status !== 401) {
+          yield* put(getZendeskToken.failure());
+        }
+      }
+      return;
+    }
+  } catch (e) {
+    yield* put(getZendeskToken.failure());
+    return;
+  }
```

Comment:
```suggestion
   if (E.isLeft(response)) {
      throw Error(readableReport(response.left));
    }
   if (response.right.status === 200) {
     yield* put(getZendeskToken.success());
     const currentValues = yield* select(sessionInfoSelector);
     yield* put(
       sessionInformationLoadSuccess(
         getOnlyNotAlreadyExistentValues(
           response.right.value,
           O.isSome(currentValues) && currentValues.value
         )
       )
     );
     return;
   }
   if (!isFastLogin || response.right.status !== 401) {
     yield* put(getZendeskToken.failure());
   }
 } catch (e) {
   yield* put(getZendeskToken.failure());
 }
```
I find this solution more readable. I removed the first `else` case because throwing an error stops the execution of the code inside the `try` block. I also removed the second `else` case by adding a `return` statement inside the preceding `if`, and then eliminated the last two `return` statements since there was no executable code following them. Do you think it will behave the same as the current implementation?

Reply:
Thanks for this suggestion! I've made the change 🚀